internal/configuration/validator/session.go

```diff
@@ -39,12 +41,16 @@ func validateSession(config *schema.SessionConfiguration, validator *schema.Stru
 		config.RememberMeDuration = schema.DefaultSessionConfiguration.RememberMeDuration // 1 month.
 	}
 
-	if config.Domain == "" {
-		validator.Push(fmt.Errorf(errFmtSessionOptionRequired, "domain"))
-	}
-
-	if strings.HasPrefix(config.Domain, "*.") {
-		validator.Push(fmt.Errorf(errFmtSessionDomainMustBeRoot, config.Domain))
+	// adding legacy session.domain to domains list.
+	if config.Domain != "" {
+		last := len(config.Domains) - 1
+		// workaround for failing tests when validator is called twice.
```

Comment:
This workaround can be avoided by rewriting the tests to not reuse the same config or by not allowing both. Until we decide on if we're allowing both this can remain as is to avoid unnecessary changes.